staging/src/k8s.io/apiserver/pkg/util/flowcontrol/metrics/metrics.go

```diff
@@ -128,14 +157,14 @@ var (
 	}
 )
 
-// UpdateFlowControlRequestsInQueue updates the value for the # of requests in the specified queues in flow control
-func UpdateFlowControlRequestsInQueue(priorityLevel string, inqueue int) {
-	apiserverCurrentInqueueRequests.WithLabelValues(priorityLevel).Set(float64(inqueue))
+// ChangeRequestsInQueues adds the given delta to the gauge of the # of requests in the queues of the specified flowSchema and priorityLevel
+func ChangeRequestsInQueues(priorityLevel, flowSchema string, delta int) {
+	apiserverCurrentInqueueRequests.WithLabelValues(priorityLevel, flowSchema).Add(float64(delta))
 }
```

Comment:
Given the usages here (except the one in `removeTimedOutRequestsFromQueueLocked`), I would prefer to expose two methods here:

```
func DecrementRequestsInQueues(priorityLevel, flowSchema string, delta int) {
    apiserverCurrentInqueueRequests.WithLabelValues(priorityLevel, flowSchema).Dec()
}	

func IncrementRequestsInQueues(priorityLevel, flowSchema string) {
    apiserverCurrentInqueueRequests.WithLabelValues(priorityLevel, flowSchema).Inc()
}	
```

Passing in a delta parameter into a method, which then applies that directly into a gauge's Add function is not super conventional, and as written, one would have to look at the method to see what's happening underneath (gauges also expose a `.Set` method, for instance).

Reply:
We can not use the `Set` method of gauges because the caller is not tracking this quantity on a per-FlowSchema basis; the caller only tracks the sum on a per-PriorityLevel basis.

I am not sure what you are alluding to where you wrote "Passing in a delta parameter into a method, which then applies that directly into a gauge's Add function is not super conventional".  What is the conventional alternative?

I do not understand the remark "one would have to look at the method to see what's happening underneath".  Are you saying a reader of this code would have to look at the implementation of gauges?  I do not see why that would be so; users of gauges can simply work with the abstraction presented by guages.  Or do you mean that a user of the `ChangeRequestsInQueues` function has to read its implementation in order to understand it?  That also does not make sense to me.  This function is intended to have a comment that describes what it does in a way that a user can understand.  If the current comment fails to accomplish that, then I think the right fix would be to the comment.

Reply:
Gauges are usually done with a `Set` but since we don't keep this quantity separately, I think using the delta is reasonable. I'm a little worried about guaranteeing correctness with this approach, but I don't see an alternative.

Reply:
Gauges have the following interface:

```
Set(float64)
Inc()
Dec()
Add(float64)
```

My main gripe is that when we wrap this method and pass in a parameter which is a float, we could either be invoking `Set` underneath or `Add` but it's not obvious which one it is (especially considering the method name `ChangeRequestsInQueues` is ambiguous). 

I'd also be okay with just changing this method name to `AddRequestsInQueues`.

Reply:
Renamed.